tools/generate_chain.py

```diff
@@ -0,0 +1,142 @@
+import cProfile
+import random
+import sqlite3
+import time
+from contextlib import closing, contextmanager
+from pathlib import Path
+from typing import Iterator, List
+
+import zstd
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.spend_bundle import SpendBundle
+from chia.util.chia_logging import initialize_logging
+from chia.util.ints import uint64
+from chia.util.path import mkdir
+from tests.block_tools import create_block_tools
+from tests.util.keyring import TempKeyring
+from tools.test_constants import test_constants
+
+
+@contextmanager
+def enable_profiler(profile: bool) -> Iterator[None]:
+    if not profile:
+        yield
+        return
+
+    with cProfile.Profile() as pr:
+        yield
+
+    pr.create_stats()
+    pr.dump_stats("generate-chain.profile")
+
+
+root_path = Path("./test-chain").resolve()
+mkdir(root_path)
+with TempKeyring() as keychain:
+
+    bt = create_block_tools(constants=test_constants, root_path=root_path, keychain=keychain)
+    initialize_logging(
+        "generate_chain", {"log_level": "DEBUG", "log_stdout": False, "log_syslog": False}, root_path=root_path
+    )
+
+    with closing(sqlite3.connect("stress-test-blockchain.sqlite")) as db:
```

Comment:
why not `:memory:`? disk reads/writes can produce inconsistent profiling results

Reply:
ok, given this is meant to generate a db, then I'd suggest 
```
PRAGMA journal_mode=WAL;
PRAGMA synchronous = 0;
PRAGMA journal_mode=MEMORY;
```
to speed things up

Reply:
The whole point of this tool is to produce this database file, it has to be saved to disk. It currently takes ~50 seconds to produce one block. The time spent writing that block to disk is negligible

Reply:
I used those pragmas initially, as well as `locking_mode=exclusive`. But being able to read the database from a separate process while this process is still writing to it is a more important feature